### PR TITLE
Update documentation, improve create/update schemas

### PR DESCRIPTION
**Pull request by ngriffiths21**
Mostly documentation updates.

I did decide to allow updating of all fields in the patient or physician profile. I think this makes the most sense because sometimes errors need to be corrected. The only one I'm torn on is whether MRN (medical record number) would ever need to be changed.

Resolves issues #2, #4, and #6.

### CODE REVIEW

File: lib/routes/physician.js [hunk 1]
```diff
@@ -10,32 +10,62 @@ module.exports = server => {
...
    *     summary: List physicians
-   *     description: Returns a list of all physicians
+   *     description: Returns a list of physicians matching query parameters.
+   *     parameters:
+   *       - in: query
+   *         name: title
+   *         schema:
+   *           type: string
+   *       - in: query
+   *         name: firstName
```
Comment (zbo14) [inline lib/routes/physician.js]: I think this should be first_name.

File: lib/routes/physician.js [hunk 2]
```diff
@@ -10,32 +10,62 @@ module.exports = server => {
...
+   *         schema:
+   *           type: string
+   *       - in: query
+   *         name: lastName
```
Comment (zbo14) [inline lib/routes/physician.js]: And this should be last_name.

File: lib/schemas/patient.js [hunk 3]
```diff
@@ -34,15 +52,20 @@ const Joi = require('joi')
  */
 
 exports.create = Joi.object({
+  MRN: Joi.string().required(),
```
Comment (zbo14) [inline lib/schemas/patient.js]: Sorry another nit, do you mind if this is `mrn`? I don't think fields are usually caps.

**Comment by zbo14** [comment]
> I did decide to allow updating of all fields in the patient or physician profile. I think this makes the most sense because sometimes errors need to be corrected. The only one I'm torn on is whether MRN (medical record number) would ever need to be changed.

Cool cool, do what you think is best for now.

Is there a format/regex for the MRN you would like to use?

File: lib/schemas/patient.js [hunk 4]
```diff
@@ -8,19 +8,37 @@ const Joi = require('joi')
...
+ *       - first_name
+ *       - last_name
  *       - dob
  *       - address
  *       - email
  *       - phone
  *     properties:
- *       MRN:
+ *       mrn:
  *         type: integer
```
Comment (zbo14) [inline lib/schemas/patient.js]: this should be string right?

File: lib/routes/patient.js [hunk 5]
```diff
@@ -13,15 +13,15 @@ module.exports = server => {
    *     description: Returns a list of patients matching the given parameters.
    *     parameters:
    *       - in: query
-   *         name: MRN
+   *         name: mrn
    *         schema:
    *           type: integer
```
Comment (zbo14) [inline lib/routes/patient.js]: string?

**Comment by zbo14** [comment]
@ngriffiths21  Also there are some discrepancies between the note create schema and the docs because of changes I made.. I'm not sure you want to address those in this PR just a heads up.

**Comment by ngriffiths21** [comment]
Might as well fix that too. The version you made looks good so the docs can
be changed. Also, yeah, mrn should be string. I probably can’t do anything
today but you can make those edits if you want.

**Pull request merged**